### PR TITLE
Feature/142866 initizalize schedules cleanly

### PR DESCRIPTION
**Pull request by butlermatt**
Some refactoring and ensuring the attributes for events are updated on load plus when events are modified/added.

### CODE REVIEW

File: lib/nodes/local_event.dart [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+    if (timeRangeString is! String) {
+      throw new Exception("Invalid Event Times");
+    }
+
+    DateTime start;
+    DateTime end;
+    Map rule;
+
+    {
```
Comment (Pacane) [inline lib/nodes/local_event.dart]: are these braces really needed?

File: lib/nodes/local_event.dart [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  @override
+  onInvoke(Map<String, dynamic> params) async {
+    var name = params[_name];
+    var timeRangeString = params[_time];
+    var ruleString = params[_rule];
+    var val = params[_value];
+
+    var p = new Path(path);
+
+    ICalendarLocalSchedule schedule = provider.getNode(p.parent.parent.parent.path);
```
Comment (Pacane) [inline lib/nodes/local_event.dart]: ayyy whoa!

File: lib/nodes/timezone.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  @override
+  onSetValue(value) {
+    if (value is String) {
+      var loc = const [
+        "UTC",
+        "Etc/GMT"
+      ].contains(value) ? TimezoneEnv.UTC : TimezoneEnv.getLocation(value);
+      if (loc != null) {
+        schedule.timezone = loc;
+        new Future(() {
```
Comment (Pacane) [inline lib/nodes/timezone.dart]: why is this wrapped in a future? Can't we just call `link.save` directly?

File: lib/nodes/timezone.dart [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          link.save();
+        });
+        schedule.loadSchedule(true);
+        return false;
+      }
+    }
+    return true;
```
Comment (Pacane) [inline lib/nodes/timezone.dart]: I'm not sure I understand the use of the boolean return here. What's the meaning?

**Review by Pacane — CHANGES_REQUESTED** [review]
I'd also dartfmt at least all your new files before merging.

File: lib/nodes/timezone.dart (hunk 4 above)
Comment (butlermatt) [reply lib/nodes/timezone.dart]: This is part of the API for SimpleNode
> Callback used to accept or reject a value when it is set. Return true to reject the value, and false to accept it.

The logic unfortunately is beyond me as to why that particular return value for that particular meaning, but it's a documented part of the dart-dslink-sdk and would be a breaking change to modify now.

**Comment by butlermatt** [comment]
I've pushed updates reflecting your comments.

**Review by Pacane — APPROVED** [review]

**Comment by butlermatt** [comment]
Thanks. leaving this pending for now until confirmation from the end user that the bug is fixed.

**Comment by butlermatt** [comment]
This contains changes that have been implemented elsewhere. Closing as duplicate.

**Pull request closed without merging**